<?xml version="1.0"?>
<launch>
  <!-- Launch file for move_base to run for one of multiple turtlebot3's -->
  
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <!-- Namespace and tf_prefix are assumed to be the same -->
  <arg name="robot_namespace"/>
  <arg name="move_forward_only" default="false"/>

  <group ns="$(arg robot_namespace)">
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
      <!-- Default configs form the TB3 repos -->
      <param name="base_local_planner" value="adwa_local_planner/ADWAPlannerROS" />
      <rosparam file="$(find multi_turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
      <rosparam file="$(find multi_turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
      <rosparam file="$(find multi_turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_turtlebot3_navigation)/param/global_costmap_params.yaml" command="load" />
      <rosparam file="$(find multi_turtlebot3_navigation)/param/move_base_params.yaml" command="load" />
      <rosparam file="$(find multi_turtlebot3_navigation)/param/adwa/adwa_local_planner_params_$(arg model).yaml" command="load" />
      <!-- <rosparam file="$(find adwa_local_planner)/param/dwa_local_planner_params.yaml" command="load" /> -->

      <!-- Set tf_prefix for frames explicity, overwriting defaults -->
      <param name="global_costmap/scan/sensor_frame" 
       value="$(arg robot_namespace)/base_scan"/>
      <param name="global_costmap/obstacle_layer/scan/sensor_frame"
	     value="$(arg robot_namespace)/base_scan"/>
      <param name="global_costmap/global_frame" 
       value="map"/>
      <param name="global_costmap/robot_base_frame"
	     value="$(arg robot_namespace)/base_footprint"/>
      <param name="local_costmap/scan/sensor_frame" 
       value="$(arg robot_namespace)/base_scan"/>
      <param name="local_costmap/obstacle_layer/scan/sensor_frame"
	     value="$(arg robot_namespace)/base_scan"/>
      <param name="local_costmap/global_frame" 
      value="$(arg robot_namespace)/odom"/>
      <param name="local_costmap/robot_base_frame"
	     value="$(arg robot_namespace)/base_footprint"/>
       
      <param name="base_global_planner" value="global_planner/GlobalPlanner" />
      <param name="planner_frequency" value="1.0" />
      <param name="planner_patience" value="5.0" />

      <param name="base_local_planner" value="adwa_local_planner/ADWAPlannerROS" />
      <param name="controller_frequency" value="5.0" />
      <param name="controller_patience" value="15.0" />

      <!-- Centralized map server -->
      <remap from="map" to="/map"/>
      <param name="DWAPlannerROS/min_vel_x" value="0.0" if="$(arg move_forward_only)" />
    </node>
  </group>
  
</launch>
